### ISSUE TITLE
Feature Request: Allow configurable assets location

### ISSUE THREAD

**Issue by bcardiff** [dillonkearns/elm-pages#315]
We are currently using elm-pages 2.0. The output of elm-pages is tweaked in some ways so we can host it under the same domain name as the main site.

The two tweaks we are doing are:

1. fingerprint the style.css, index.js, elm.js
2. change its location from /* to /assets/*

In elm-pages 3.0 the fingerprint is already covered AFAIK. But I would be interested in pushing a built-in configurable location to avoid post build tweaks of the output. Tweaking the assets location requires changing its references in html pages which is not ideal.

Using a different base path for the whole project is not enough because we still what page at the root of the site built by elm-pages.

Some scenarios that are unblocked by a custom asset location are

a. embed the elm-pages output in any site, since you can choose any free path for the elm-pages asset
b. split a site between multiple elm-pages projects (as long as they don't clash in paths)
c. allow cache rules by routes and not only by file extensions.

**Comment by dillonkearns** [comment]
Hey @bcardiff, thank you for sharing that feedback!

You're correct that elm-pages v3 does fingerprinting.

The default location for outputting assets is `/assets` by default with v3 as well since that's the configuration that ViteJS uses (and assets are processed through ViteJS).

You can also customize the configuration of the output path the same way you would in a ViteJS project like this: https://stackoverflow.com/a/71190586.

I actually had to make a little tweak to make this work, but I've now published it as the latest beta package, which you can try out with this starter repo: https://github.com/dillonkearns/elm-pages-3-alpha-starter.

I believe this gives you quite a lot of control over the asset output (even changing the filename format and where the hash shows up for example), so I think this meets all the needs your describing here (thanks to leveraging the mature ViteJS toolchain). Let me know how it looks to you!

**Comment by dillonkearns** [comment]
I'll close this for now, but feel free to reopen if there's anything further to discuss. Thanks!
